php/functions.php

```diff
@@ -0,0 +1,151 @@
+ *
+ * @param string $validatedProjectName is the new string pulled from the admin.php 'project name' text input
+ * @param string $validatedImageUrl is the new string pulled from the admin.php 'upload project image' text input
+ * @param string $validatedProjectUrl is the new string pulled from the admin.php 'upload project' text input
+ * @param string $validatedProjectSummary is the new string pulled from the admin.php 'project summary' text input
+ * @param object $db is the database being updated by the function
+ *
+ */
+function uploadProject(string $validatedProjectName, string $validatedImageUrl, string $validatedProjectUrl, string $validatedProjectSummary, PDO $db) : array {
+    $uploadProjectQuery = $db->prepare("INSERT INTO `projects`(`name`, `image_url`, `url`, `summary`) VALUES (?,?,?,?);");
+    $uploadProjectQuery->execute([$validatedProjectName, $validatedImageUrl, $validatedProjectUrl, $validatedProjectSummary]);
+}
+
+/*
+ * This function hides the selected project in the table `projects` defined in the drop down menu of the admin.php page
+ *
+ * @param string $projectName is the new string pulled from the admin.php 'project name' text input
+ * @param object $db is the database being updated by the function
+ *
+ */
+function hideProject(string $projectId, PDO $db) {
+    $hideProjectQuery = $db->prepare("UPDATE `projects` SET `deleted` = 1 WHERE `id`= ?;");
+    $hideProjectQuery->execute([$projectId]);
+}
+
+/*
+ * This function retrieves the data defined in the inset query from the 'projects' table of the 'portfolio' database
+ *
+ * @param PDO $db is the database the data is being pulled from
+ *
+ * @return array $projectResults is an associative array of the data fetched by $projectQuery
+ */
+function getDbProject(PDO $db) : array {
+    $projectQuery = $db->prepare("SELECT `name`, `image_url`, `url`, `summary` FROM `projects` WHERE `deleted` = 0;");
+    $projectQuery->execute();
+    $projectResults = $projectQuery->fetchAll();
+    return $projectResults;
+}
+
+/*
+ * This function uploads all data from the 'projects' table of the 'portfolio' database to index.php in the
+ * projects section of the page
+ *
+ * @param array $projectResults represents all 'projects' held in the 'project' table of the 'portfolio' database
+ *
+ * @return array $result represents an individual project that is displayed dynamically through the foreach command
+ */
+function displayProjects(array $projectResults) : string
+{
+    $result = '';
+    foreach ($projectResults as $projectResult) {
+        if (array_key_exists('url', $projectResult) && array_key_exists('image_url', $projectResult) && array_key_exists('summary', $projectResult)) {
+            $result .= '<div class="projectOne">
+            <div class="projectBox">
+                <a href=' . $projectResult['url'] . '><img class="projectPics" src=' . $projectResult['image_url'] . '></a>
```

Comment:
the urls in your db just contain the image name. but the images are stored in images/ so your images dont actually work